service/src/test/java/bio/terra/workspace/service/spendprofile/SpendProfileServiceTest.java

```diff
@@ -2,61 +2,186 @@
 
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
+import bio.terra.profile.api.ProfileApi;
+import bio.terra.profile.client.ApiException;
+import bio.terra.profile.model.ProfileModel;
+import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
 import bio.terra.workspace.app.configuration.external.SpendProfileConfiguration;
-import bio.terra.workspace.common.BaseConnectedTest;
-import bio.terra.workspace.connected.UserAccessUtils;
-import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.iam.model.SamConstants;
+import bio.terra.workspace.service.spendprofile.client.BpmClientProvider;
+import bio.terra.workspace.service.spendprofile.exceptions.BillingProfileManagerServiceAPIException;
 import bio.terra.workspace.service.spendprofile.exceptions.SpendUnauthorizedException;
 import com.google.common.collect.ImmutableList;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.UUID;
 import org.junit.jupiter.api.Test;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 
-class SpendProfileServiceTest extends BaseConnectedTest {
-  @Autowired SamService samService;
-  @Autowired SpendProfileConfiguration spendProfileConfiguration;
-  @Autowired SpendConnectedTestUtils spendUtils;
-  @Autowired UserAccessUtils userAccessUtils;
+class SpendProfileServiceTest extends BaseUnitTest {
```

Comment:
I changed this over to a unit test (vs. connected) to ease testing. That is, this test was making a live call to Sam and expecting spend profile resources to be present. Updating this to mock out a sam response seemed like the better way to ease development.